### PR TITLE
OXT-1056: Package both C and OCaml xenstored. Default to C.

### PR DESCRIPTION
**Pull request by dozylynx**
This PR for OXT-1056 builds upon the related PR #604 for OXT-1067 (to unfork the OCaml libraries), so  includes those commits. This PR is fine to go in and the other just be retired.

There's a minor interaction between the OE standard behaviour for update-rc.d and update-alternatives that becomes evident when packaging alternative versions of the same init script, so there's a workaround for that in this PR, in the xenclient-dom0-image.bb section. OE also needed more convincing than would be expected to only install a single xenstored package, so that's enforced in the same file.

The results are correct: C daemon installed and working; OCaml daemon packaged but not installed.

Signed-off-by: Christopher Clark <christopher.clark6@baesystems.com>

### CODE REVIEW

File: recipes-openxt/xenclient/xenclient-toolstack_git.bb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 DESCRIPTION = "XenClient toolstack"
 LICENSE = "LGPLv2.1"
 LIC_FILES_CHKSUM = "file://COPYING;md5=321bf41f280cf805086dd5a720b37785"
-DEPENDS += "ocaml-cross ocaml-dbus ocaml-camomile xen xz"
+DEPENDS += " ocaml-cross ocaml-dbus ocaml-camomile xen xen-libxl xz "
```
Comment (eric-ch) [inline recipes-openxt/xenclient/xenclient-toolstack_git.bb]: leading and trailing spaces are not required ([+=/=+, appending with spaces](https://www.yoctoproject.org/docs/1.6/bitbake-user-manual/bitbake-user-manual.html#appending-and-prepending)).

File: recipes-openxt/xenclient/xenclient-toolstack_git.bb [hunk 2]
```diff
@@ -46,11 +46,16 @@ S = "${WORKDIR}/git"
...
+
+OCAML_INSTALL_LIBS = " \
+    libs/uuid \
+    libs/stdext \
+    libs/json \
+    libs/jsonrpc \
+    libs/http \
+    libs/log \
+    libs/common \
+    "
```
Comment (eric-ch) [inline recipes-openxt/xenclient/xenclient-toolstack_git.bb]: A couple of libs are missing. They are required to build uid:
```
--- a/recipes-openxt/xenclient/xenclient-toolstack_git.bb
+++ b/recipes-openxt/xenclient/xenclient-toolstack_git.bb
@@ -50,10 +50,13 @@ do_compile() {
 OCAML_INSTALL_LIBS = " \
     libs/uuid \
     libs/stdext \
+    libs/mmap \
     libs/json \
     libs/jsonrpc \
     libs/http \
     libs/log \
+    libs/xc libs/eventchn \
+    libs/xb libs/xs \
     libs/common \
     "
```

File: recipes-openxt/xenclient/xenclient-toolstack_git.bb (hunk 1 above)
Comment (dozylynx) [inline recipes-openxt/xenclient/xenclient-toolstack_git.bb]: ack, thanks - fixed.

File: recipes-openxt/xenclient/xenclient-toolstack_git.bb (hunk 2 above)
Comment (dozylynx) [inline recipes-openxt/xenclient/xenclient-toolstack_git.bb]: The change here is deliberate. Those five libraries are no longer part of the toolstack build as the code has been removed. Upstream versions are packaged by the xen-libxl recipe instead and uid obtains copies of those libraries from the xen-ocaml-libs package which was introduced by the preceding commit in this PR -- that's why the uid recipe gains a DEPENDS on xen-libxl in this change.

**Comment by eric-ch** [comment]
depends on:
- https://github.com/OpenXT/uid/pull/3,
- https://github.com/OpenXT/toolstack/pull/28,
- https://github.com/OpenXT/manager/pull/79,
afaict.

File: recipes-openxt/xenclient/xenclient-toolstack_git.bb (hunk 2 above)
Comment (eric-ch) [inline recipes-openxt/xenclient/xenclient-toolstack_git.bb]: Ahh I did not see the relevant PRs in the other repositories, my bad!

**Comment by eric-ch** [comment]
Build [760](http://openxt-builder.ainfosec.com:8010/builders/openxt/builds/760), including these changes, is still going but already built xen-libxl.
I built it and installed it on a Dell E6440, ```/usr/sbin/xenstored.xen-xenstored-c``` is installed and running.
LGTM, +1.

**Comment by rossphilipson** [comment]
Are all of them ready to go in including the 3 this depends on?

**Comment by dozylynx** [comment]
hi @rossphilipson : not sure if you're asking me but if you are: yes.

**Pull request merged**